DataFormats/L1TParticleFlow/interface/PFCandidate.h

```diff
@@ -0,0 +1,61 @@
+#ifndef DataFormats_L1TParticleFlow_PFCandidate_h
+#define DataFormats_L1TParticleFlow_PFCandidate_h
+
+#include <vector>
+#include "DataFormats/L1Trigger/interface/L1Candidate.h"
+#include "DataFormats/L1Trigger/interface/Muon.h"
+#include "DataFormats/L1TParticleFlow/interface/PFCluster.h"
+#include "DataFormats/L1TParticleFlow/interface/PFTrack.h"
+
+namespace l1t {
+
+  class PFCandidate : public L1Candidate {
+  public:
+    typedef edm::Ptr<l1t::Muon> MuonRef;
+    enum ParticleType { ChargedHadron = 0, Electron = 1, NeutralHadron = 2, Photon = 3, Muon = 4 };
+
+    PFCandidate() {}
+    PFCandidate(ParticleType kind,
+                int charge,
+                const LorentzVector& p,
+                float puppiWeight = -1,
+                int hwpt = 0,
+                int hweta = 0,
+                int hwphi = 0)
+        : PFCandidate(kind, charge, PolarLorentzVector(p), puppiWeight, hwpt, hweta, hwphi) {}
+    PFCandidate(ParticleType kind,
+                int charge,
+                const PolarLorentzVector& p,
+                float puppiWeight = -1,
+                int hwpt = 0,
+                int hweta = 0,
+                int hwphi = 0);
+
+    ParticleType id() const { return ParticleType(hwQual()); }
+
+    const PFTrackRef& pfTrack() const { return trackRef_; }
+    void setPFTrack(const PFTrackRef& ref) { trackRef_ = ref; }
+
+    const PFClusterRef& pfCluster() const { return clusterRef_; }
+    void setPFCluster(const PFClusterRef& ref) { clusterRef_ = ref; }
+
+    const MuonRef& muon() const { return muonRef_; }
+    void setMuon(const MuonRef& ref) { muonRef_ = ref; }
+
+    /// PUPPI weight (-1 if not available)
+    float puppiWeight() const { return puppiWeight_; }
```

Comment:
why you use a different approach for `puppiWeight` instead of something like
```
    const float& puppiWeight() const { return puppiWeight; }
    void setpuppiWeight(const float& val) { puppiWeight_ = puppiWeight; }
```
?

Reply:
well I see @makortel suggested the opposite here
https://github.com/cms-sw/cmssw/pull/29346#discussion_r400587009
Anyhow, please add setPuppiWeight, if you think it can be useful.

Reply:
Because it is more efficient to return float by value.
Returning primitive types by reference is less efficient because it saves nothing and adds overhead.

Reply:
`puppiWeight_ ` is set in constructor. 
  /// PUPPI weight (-1 if not available)
Not allowing change of weight through setter.